test/common/upstream/health_check_fuzz.cc

```diff
@@ -0,0 +1,200 @@
+#include "test/common/upstream/health_check_fuzz.h"
+
+#include <chrono>
+
+#include "test/common/upstream/utility.h"
+#include "test/fuzz/utility.h"
+
+namespace Envoy {
+namespace Upstream {
+
+void HealthCheckFuzz::allocHealthCheckerFromProto(
+    const envoy::config::core::v3::HealthCheck& config) {
+  health_checker_ = std::make_shared<TestHttpHealthCheckerImpl>(
+      *cluster_, config, dispatcher_, runtime_, random_,
+      HealthCheckEventLoggerPtr(event_logger_storage_.release()));
+  ENVOY_LOG_MISC(trace, "Created Test Health Checker");
+}
+
+void HealthCheckFuzz::initializeAndReplay(test::common::upstream::HealthCheckTestCase input) {
+  try {
+    allocHealthCheckerFromProto(input.health_check_config());
+  } catch (EnvoyException& e) {
+    ENVOY_LOG_MISC(debug, "EnvoyException: {}", e.what());
+    return;
+  }
+  ON_CALL(runtime_.snapshot_, featureEnabled("health_check.verify_cluster", 100))
+      .WillByDefault(testing::Return(input.http_verify_cluster()));
+  cluster_->prioritySet().getMockHostSet(0)->hosts_ = {
+      makeTestHost(cluster_->info_, "tcp://127.0.0.1:80")};
+  expectSessionCreate();
+  expectStreamCreate(0);
+  if (input.start_failed()) {
+    cluster_->prioritySet().getMockHostSet(0)->hosts_[0]->healthFlagSet(
+        Host::HealthFlag::FAILED_ACTIVE_HC);
+  }
+  health_checker_->start();
+  ON_CALL(runtime_.snapshot_, getInteger("health_check.min_interval", _))
+      .WillByDefault(testing::Return(45000));
+  // If has an initial jitter, this calls onIntervalBase and finishes startup
+  if (input.health_check_config().initial_jitter().seconds() != 0) {
```

Comment:
if this is a protobuf duration, is there any possibility that the nanosecond field could be set high and seconds could be 0?